config/routes.rb

```diff
@@ -1,10 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
-  # Can be used by load balancers and uptime monitors to verify that the app is live.
-  get "up" => "rails/health#show", as: :rails_health_check
-
-  # Defines the root path route ("/")
-  # root "posts#index"
+  root "users#index"
+  get "users/:id" => "users#user", as: :user
+  get "users/:id/posts" => "users#posts", as: :posts
+  get "users/:id/posts/:p_id" => "users#post", as: :post
```

Comment:
- [ ] Kindly define your routes correctly. The users and posts routes should only have `index` and `show` routes. We should not have `users#user`, `users#posts` or `users#post` route. We should only have `users#index`, `users#show`, `posts#index` and `posts#show`, which we can define using the following.

  ```
  root 'users#index'
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```